### PR TITLE
Implementing: collatz-conjecture

### PR DESCRIPTION
**Pull request by Average-user**
This is on the list of Unimplemented exercises of Scala in **exercism.io**

### CODE REVIEW

File: config.json [hunk 1]
```diff
@@ -47,6 +47,16 @@
       ]
     },
     {
+      "uuid": "65a76aba-a485-222d-84ba-e9a445b25be6",
+      "slug": "collatz-conjecture",
+      "core": true,
+      "unlocked_by": null,
+      "difficulty": 1,
+      "topics": [
+        "Recursion"
```
Comment (ricemery) [inline config.json]: Could you add "Mathematics" as a topic?

File: config.json [hunk 2]
```diff
@@ -47,6 +47,16 @@
       ]
     },
     {
+      "uuid": "65a76aba-a485-222d-84ba-e9a445b25be6",
+      "slug": "collatz-conjecture",
+      "core": true,
+      "unlocked_by": null,
+      "difficulty": 1,
```
Comment (ricemery) [inline config.json]: Maybe change the difficulty to 2? 1 seems a tad low.
And, maybe change unlocked_by to "leap"

File: exercises/collatz-conjecture/src/test/scala/CollatzConjectureTest.scala [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+  test("Large number of even and odd steps") {
+    pending
+    CollatzConjecture.collatz(1000000) should be (152)
+  }
+
+  test("Large number of even and odd steps") {
+    pending
+    CollatzConjecture.collatz(1000000) should be (152)
+  }
```
Comment (ricemery) [inline exercises/collatz-conjecture/src/test/scala/CollatzConjectureTest.scala]: Missing final "}". This class does not compile.
I am actually surprised that the travis CI build did not error.

As a note - we are going through the process of creating test code generators to take the exercism canonical-data.xml file and generate the unit tests. No big deal in this case. Ultimately I can build the generator for this exercise once I get some time. For now, hand generated unit tests are fine. If you are interested, the test generators are in the scala/testgen directory.

**Comment by Average-user** [comment]
@ricemery  I'll make the corrections. Thanks

**Comment by ricemery** [comment]
@Average-user  - Sorry for the delay. I didn't see that you had submitted your changes.
I made some additional changes, and my lack of git skills resulted in a new pull request. The changes were 

* Moved example.scala into root of the exercise. We keep the example code out of the src directory.
* Changed the signature of the function to expect an Option[Int]. This change was to allow for clean handling of the error cases.
* Removed duplicate test case. Added the two error test cases.

**Comment by Average-user** [comment]
@ricemery So what will happen with this PR?

**Comment by ricemery** [comment]
I think we close this pull request. And merge the new one.

**Comment by Average-user** [comment]
Works for me

**Comment by ricemery** [comment]
Thanks for working on this @Average-user !

**Comment by Average-user** [comment]
No problem. You ended up doing more, probably.

**Comment by ricemery** [comment]
Not at all. You did the hard part. I had the luxury of tweaking an already
working solution.
We can use all the help we can get on the Scala track.

**Pull request merged**